caffe2/serialize/inline_container.h

```diff
@@ -141,18 +149,25 @@ class CAFFE2_API PyTorchStreamWriter final {
   ~PyTorchStreamWriter();
 
  private:
+  void setup(const string& file_name);
   void valid(const char* what, const char* info = "");
   size_t current_pos_ = 0;
   std::unique_ptr<mz_zip_archive> ar_;
   std::string archive_name_;
-  std::ostream* out_;
+  std::ostream* out_ = nullptr;
```

Comment:
Is it really providing any value to have both paths?  Could we not just convert all callers to use the `std::function` path?

Reply:
That sounds good, I'll unify behind std::function<>.